week5/week5.py

```diff
@@ -0,0 +1,116 @@
+class TrieNode:
+    def __init__(self):
+        # one extra character for the apostrophe
+        self.links = [0] * 27
+        self.isEnd = False
+
+class Dictionary:
+    def __init__(self):
+        self.trie = TrieNode()
+        f = open("words2.txt", "r")
```

Comment:
I think it would be better not to hard code this, so ideally make it take a `List[str]`.

This also makes testing a lot clearer, since you can specify the Dictionary words in the test case. Otherwise it depends on what's in `words2.txt`, which makes it a lot less easy to work out the test.